### ISSUE TITLE
conda install local FAILS using python3 but not python2

### ISSUE THREAD

**Issue by mikecormier** [conda/conda#8472]
## Current Behavior

Using 'conda install --use-local` to install and test a local conda package, the installation fails while using python3, however, no problem occurs while using python2. 

`conda build` works and correctly build the package using both python2 and 3. 

The problem seems to be with both conda 4.6.8 and 4.6.9. 

### Steps to Reproduce

```
conda build <conda recipe>
conda install -v --use-local -y <local conda package>
```
### Output:
```
conda install -v --use-local -y gaps

Collecting package metadata: ...working... done
Solving environment: ...working... done
initializing UnlinkLinkTransaction with
  target_prefix: /anaconda2/envs/env-python3
  unlink_precs:
    
  link_precs:
    local::gaps-1-0



## Package Plan ##

  environment location: /anaconda2/envs/env-python3

  added / updated specs:
    - gaps


The following packages will be downloaded:

    package        |            build
    -------------|-----------------
    gaps-1           |                0           6 KB  local
    -------------------------------
                                           Total:           6 KB

The following NEW packages will be INSTALLED:

  gaps anaconda2/conda-bld/noarch::gaps-1-0


Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... ===> LINKING PACKAGE: local::gaps-1-0 <===
  prefix=/anaconda2/envs/env-python3
  source=//anaconda2/pkgs/gaps-1-0


$ bash /scratch/local/u1138933/tmpYAdva5
==> cwd: /anaconda2/envs/env-python3/bin <==
==> exit code: 1 <==
==> stdout <==

==> stderr <==



An error occurred while installing package 'local::gaps1-1-0'.
LinkError: post-link script failed for package local::gaps-1-0
running your command again with `-v` will provide additional information
location of failed script: /anaconda2/envs/env-python3/bin/.gaps-post-link.sh
==> script messages <==
<None>

Attempting to roll back.


===> REVERSING PACKAGE LINK: local::gaps-1-0 <===
  prefix=/anaconda2/envs/env-python3


failed
An error occurred while installing package 'local::gap-1-0'.
LinkError: post-link script failed for package local::gaps-1-0
running your command again with `-v` will provide additional information
location of failed script: /anaconda2/envs/env-python3/bin/.gaps-post-link.sh
==> script messages <==
<None>

Attempting to roll back.


Rolling back transaction: ...working... done
Traceback (most recent call last):
  File "/anaconda2/lib/python2.7/site-packages/conda/exceptions.py", line 1003, in __call__
    return func(*args, **kwargs)
  File "/anaconda2/lib/python2.7/site-packages/conda/cli/main.py", line 84, in _main
    exit_code = do_call(args, p)
  File "/anaconda2/lib/python2.7/site-packages/conda/cli/conda_argparse.py", line 82, in do_call
    exit_code = getattr(module, func_name)(args, parser)
  File "/anaconda2/lib/python2.7/site-packages/conda/cli/main_install.py", line 20, in execute
    install(args, parser, 'install')
  File "/anaconda2/lib/python2.7/site-packages/conda/cli/install.py", line 275, in install
    handle_txn(unlink_link_transaction, prefix, args, newenv)
  File "/anaconda2/lib/python2.7/site-packages/conda/cli/install.py", line 304, in handle_txn
    unlink_link_transaction.execute()
  File "/anaconda2/lib/python2.7/site-packages/conda/core/link.py", line 241, in execute
    self._execute(tuple(concat(interleave(itervalues(self.prefix_action_groups)))))
  File "/anaconda2/lib/python2.7/site-packages/conda/core/link.py", line 587, in _execute
    rollback_excs,
CondaMultiError: post-link script failed for package local::gaps-1-0
running your command again with `-v` will provide additional information
location of failed script: /anaconda2/envs/env-python3/bin/.gaps-post-link.sh
==> script messages <==
<None>
```

## Expected Behavior

`conda install --use-local` should work for both python 2 and 3

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
     active environment : env-python3
    active env location : /anaconda2/envs/env-python3
            shell level : 2
       user config file : /home/.condarc
 populated config files : /home/.condarc
          conda version : 4.6.9
    conda-build version : 3.17.8
         python version : 2.7.15.final.0
       base environment : /anaconda2  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://conda.anaconda.org/bioconda/linux-64
                          https://conda.anaconda.org/bioconda/noarch
                          https://conda.anaconda.org/ggd-genomics/linux-64
                          https://conda.anaconda.org/ggd-genomics/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /anaconda2/pkgs
                          /home/.conda/pkgs
       envs directories : /anaconda2/envs
                          home/.conda/envs
               platform : linux-64
             user-agent : conda/4.6.9 requests/2.21.0 CPython/2.7.15 Linux/3.10.0-957.1.3.el7.x86_64 centos/7.6.1810 glibc/2.17
                UID:GID : 1218369:2294
             netrc file : None
           offline mode : False
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
==> /home/.condarc <==
channels:
  - conda-forge
  - bioconda
  - ggd-genomics
  - defaults
allow_conda_downgrades: True
```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
# packages in environment at /anaconda2/envs/env-python3:
#
# Name                    Version                   Build  Channel
aiofiles                  0.4.0                   py_1001    conda-forge
aioftp                    0.12.0                     py_0    conda-forge
aiohttp                   3.4.4           py36h14c3975_1000    conda-forge
alabaster                 0.7.12                     py_0    conda-forge
anaconda-client           1.6.14                     py_0    conda-forge
argh                      0.26.2                py36_1001    conda-forge
asn1crypto                0.24.0                py36_1003    conda-forge
async-timeout             3.0.1                   py_1000    conda-forge
atomicwrites              1.3.0                      py_0    conda-forge
attrs                     19.1.0                     py_0    conda-forge
babel                     2.6.0                      py_1    conda-forge
backoff                   1.6.0                      py_1    conda-forge
beautifulsoup4            4.6.3                 py36_1000    conda-forge
bioconda-utils            0.15.10                  pypi_0    pypi
blas                      1.0                         mkl    defaults
boltons                   18.0.1                     py_0    conda-forge
boto3                     1.9.113                    py_0    conda-forge
botocore                  1.12.113                   py_0    conda-forge
bzip2                     1.0.6             h14c3975_1002    conda-forge
ca-certificates           2019.3.9             hecc5488_0    conda-forge
cairo                     1.16.0            ha4e643d_1000    conda-forge
certifi                   2019.3.9                 py36_0    conda-forge
cffi                      1.12.2           py36h9745a5d_0    conda-forge
chardet                   3.0.4                 py36_1003    conda-forge
check-sort-order          0.0.3                         1    bioconda
clyent                    1.2.2                      py_1    conda-forge
colorlog                  3.1.4                 py36_1000    conda-forge
conda                     4.6.8                    py36_0    conda-forge
conda-build               3.17.8                   py36_0    conda-forge
conda-env                 2.6.0                         1    conda-forge
cryptography              2.6.1            py36h9d9f1b6_0    conda-forge
curl                      7.64.0               h646f8bb_2    conda-forge
decorator                 4.3.2                      py_0    conda-forge
docutils                  0.14                  py36_1001    conda-forge
expat                     2.2.5             hf484d3e_1002    conda-forge
filelock                  3.0.10                     py_0    conda-forge
fontconfig                2.13.1            h2176d3f_1000    conda-forge
freetype                  2.9.1             h94bbf69_1005    conda-forge
future                    0.17.1                py36_1000    conda-forge
fuzzywuzzy                0.17.0                     py_0    conda-forge
galaxy-lib                18.9.2                     py_0    bioconda
gettext                   0.19.8.1          h9745a5d_1001    conda-forge
ggd                       0.0.8                     dev_0    <develop>
gidgethub                 3.0.0                      py_0    conda-forge
git                       2.21.0          pl526h2882143_0    conda-forge
gitdb2                    2.0.5                      py_0    conda-forge
gitpython                 2.1.11                     py_0    conda-forge
glib                      2.58.3            hf63aee3_1001    conda-forge
glob2                     0.6                        py_0    conda-forge
graphite2                 1.3.13            hf484d3e_1000    conda-forge
graphviz                  2.38.0            hf68f40c_1011    conda-forge
grch37-esp-variants       1                       pl526_0    file:///anaconda2/envs/env-python3/conda-bld
grch37-reference-genome   1                             3    ggd-genomics
grch38-reference-genome-ensembl 1                             3    ggd-genomics
gsort                     0.0.6                         1    bioconda
harfbuzz                  2.3.1                h6824563_0    conda-forge
htslib                    1.9                  ha228f0b_6    bioconda
icu                       58.2              hf484d3e_1000    conda-forge
idna                      2.7                   py36_1002    conda-forge
idna_ssl                  1.1.0                 py36_1000    conda-forge
imagesize                 1.1.0                      py_0    conda-forge
intel-openmp              2019.1                      144    defaults
involucro                 1.1.2                         1    bioconda
ipython_genutils          0.2.0                      py_1    conda-forge
jinja2                    2.10                       py_1    conda-forge
jmespath                  0.9.4                      py_0    conda-forge
jpeg                      9c                h14c3975_1001    conda-forge
jsonschema                2.6.0                 py36_1002    conda-forge
jupyter_core              4.4.0                      py_0    conda-forge
krb5                      1.16.3            h05b26f9_1001    conda-forge
libarchive                3.3.3             he5c8681_1003    conda-forge
libcurl                   7.64.0               h541490c_2    conda-forge
libdeflate                1.0                  h14c3975_1    bioconda
libedit                   3.1.20170329      hf8c457e_1001    conda-forge
libffi                    3.2.1             hf484d3e_1005    conda-forge
libgcc-ng                 7.3.0                hdf63c60_0    conda-forge
libgfortran-ng            7.2.0                hdf63c60_3    conda-forge
libiconv                  1.15              h14c3975_1004    conda-forge
libidn2                   2.1.1                h14c3975_0    conda-forge
liblief                   0.9.0                h1532aa0_0    conda-forge
libpng                    1.6.36            h84994c4_1000    conda-forge
libssh2                   1.8.0             h90d6eec_1004    conda-forge
libstdcxx-ng              7.3.0                hdf63c60_0    conda-forge
libtiff                   4.0.10            h648cc4a_1001    conda-forge
libtool                   2.4.6             h14c3975_1002    conda-forge
libunistring              0.9.10               h14c3975_0    conda-forge
libuuid                   2.32.1            h14c3975_1000    conda-forge
libxcb                    1.13              h14c3975_1002    conda-forge
libxml2                   2.9.8             h143f9aa_1005    conda-forge
lz4-c                     1.8.1.2                       0    conda-forge
lzo                       2.10              h14c3975_1000    conda-forge
markupsafe                1.1.1            py36h14c3975_0    conda-forge
mkl                       2019.1                      144    defaults
mkl_fft                   1.0.10           py36h14c3975_1    conda-forge
mkl_random                1.0.2            py36h637b7d7_2    conda-forge
more-itertools            4.3.0                 py36_1000    conda-forge
multidict                 4.5.2           py36h14c3975_1000    conda-forge
nbformat                  4.4.0                      py_1    conda-forge
ncurses                   6.1               hf484d3e_1002    conda-forge
networkx                  1.11                     py36_0    conda-forge
numpy                     1.15.4           py36h7e9f1db_0    defaults
numpy-base                1.15.4           py36hde5b4d6_0    defaults
openblas                  0.3.3             h9ac9557_1001    conda-forge
openssl                   1.1.1b               h14c3975_1    conda-forge
oyaml                     0.7                   py36_1000    conda-forge
packaging                 19.0                       py_0    conda-forge
pandas                    0.23.4          py36h637b7d7_1000    conda-forge
pango                     1.40.14           h4ea9474_1004    conda-forge
patchelf                  0.9               hf484d3e_1003    conda-forge
pcre                      8.41              hf484d3e_1003    conda-forge
perl                      5.26.2            h14c3975_1002    conda-forge
pip                       19.0.3                   py36_0    conda-forge
pixman                    0.34.0            h14c3975_1003    conda-forge
pkginfo                   1.5.0.1                    py_0    conda-forge
pluggy                    0.9.0                      py_0    conda-forge
psutil                    5.5.1            py36h14c3975_0    conda-forge
pthread-stubs             0.4               h14c3975_1001    conda-forge
py                        1.8.0                      py_0    conda-forge
py-lief                   0.9.0            py36hf484d3e_0    conda-forge
pyaml                     17.12.1                    py_0    conda-forge
pycosat                   0.6.3           py36h14c3975_1001    conda-forge
pycparser                 2.19                       py_0    conda-forge
pydotplus                 2.0.2                      py_2    conda-forge
pygithub                  1.34                     py36_0    conda-forge
pygments                  2.3.1                      py_0    conda-forge
pyjwt                     1.7.1                      py_0    conda-forge
pyopenssl                 19.0.0                   py36_0    conda-forge
pyparsing                 2.3.1                      py_0    conda-forge
pysocks                   1.6.8                 py36_1002    conda-forge
pytest                    4.3.1                    py36_0    conda-forge
python                    3.6.7             h381d211_1004    conda-forge
python-dateutil           2.8.0                      py_0    conda-forge
python-levenshtein        0.12.0          py36h14c3975_1000    conda-forge
python-libarchive-c       2.8                   py36_1004    conda-forge
pytz                      2018.9                     py_0    conda-forge
pyyaml                    3.13            py36h14c3975_1001    conda-forge
readline                  7.0               hf8c457e_1001    conda-forge
regex                     2018.08.29      py36h14c3975_1000    conda-forge
requests                  2.21.0                py36_1000    conda-forge
ruamel_yaml               0.15.71         py36h14c3975_1000    conda-forge
s3transfer                0.2.0                    py36_0    conda-forge
samtools                  1.9                 h8571acd_11    bioconda
setuptools                40.8.0                   py36_0    conda-forge
simplejson                3.8.1                    py36_0    bioconda
six                       1.11.0                py36_1001    conda-forge
smmap2                    2.0.5                      py_0    conda-forge
snowballstemmer           1.2.1                      py_1    conda-forge
sphinx                    1.8.4                    py36_0    conda-forge
sphinx_rtd_theme          0.4.3                      py_0    conda-forge
sphinxcontrib-websupport  1.1.0                      py_1    conda-forge
sqlite                    3.26.0            h67949de_1001    conda-forge
tk                        8.6.9             h84994c4_1000    conda-forge
tqdm                      4.31.1                     py_0    conda-forge
traitlets                 4.3.2                 py36_1000    conda-forge
uritemplate               3.0.0                      py_1    conda-forge
urllib3                   1.23                  py36_1001    conda-forge
vt                        2015.11.10           he941832_3    bioconda
wget                      1.20.1               h90d6eec_0    conda-forge
wheel                     0.33.1                   py36_0    conda-forge
xorg-kbproto              1.0.7             h14c3975_1002    conda-forge
xorg-libice               1.0.9             h14c3975_1004    conda-forge
xorg-libsm                1.2.3             h4937e3b_1000    conda-forge
xorg-libx11               1.6.7             h14c3975_1000    conda-forge
xorg-libxau               1.0.9                h14c3975_0    conda-forge
xorg-libxdmcp             1.1.2             h14c3975_1007    conda-forge
xorg-libxext              1.3.3             h14c3975_1004    conda-forge
xorg-libxpm               3.5.12            h14c3975_1002    conda-forge
xorg-libxrender           0.9.10            h14c3975_1002    conda-forge
xorg-libxt                1.1.5             h14c3975_1002    conda-forge
xorg-renderproto          0.11.1            h14c3975_1002    conda-forge
xorg-xextproto            7.3.0             h14c3975_1002    conda-forge
xorg-xproto               7.0.31            h14c3975_1007    conda-forge
xz                        5.2.4             h14c3975_1001    conda-forge
yaml                      0.1.7             h14c3975_1001    conda-forge
yarl                      1.3.0           py36h14c3975_1000    conda-forge
zlib                      1.2.11            h14c3975_1004    conda-forge
zstd                      1.3.3                         1    conda-forge
```
</p></details>

**Comment by msarahan** [comment]
what is in your post-link script there?

/anaconda2/envs/env-python3/bin/.gaps-post-link.sh

**Comment by mikecormier** [comment]
@msarahan, I figured out the problem. conda failed to install because I had two local builds of the same package, one in conda-bld/noarch and one in conda-bld/linux-64. When either one of the two was removed the local package installed just fine. I'm not sure why I kept getting the error I was getting, but a feature that could be implemented in the future is to identify this redundancy and either inform the user to remove one or choose one.
